URIs.md

```diff
@@ -0,0 +1,103 @@
+# Farcaster URIs
+
+Farcaster Messages may reference other messages, online content, or on-chain activity. An extension of the URI standard is employed by Farcaster to identify all resources and activities that can be referenced within protocol messages. 
+
+## Requirements
+
+- *Extensible*: The standard should anticipate the addition of new resource types.
+- *Versionable*: The standard should accommodate new and evolving requirements for existing resource types while preserving support for older identifiers
+- *Forward Compatible*: The addition of new resource and activity types to the standard should not require a coordinated upgrade of all Farcaster nodes and clients. The standard should allow for clear differentiation between invalidly constructed identifiers and identifiers that employ the use of not-yet-supported resource types.
+
+*Non-Requirements*
+
+- *Canonical Representation*: The protocol need not require all parties produce identical identifiers for the same resource. Canonicalization is necessary where different parties must independently produce identical representations of the same object (as when used for hash verification), and assists equality checking by allowing comparison of canonical identifiers. Certain resources, especially within the web, cannot be coerced into using a single canonical identifier. Nonetheless, the standard should minimize the space of valid identifiers for a single resource and should impose unique identifiers where possible.
+
+## Specification
+
+The standard is an extension of the common [URL standard](https://datatracker.ietf.org/doc/html/rfc1738) with support for the HTTP, HTTPS, and [IPFS](https://docs.ipfs.io/how-to/address-ipfs-on-web/#native-urls) protocols as well as the newly introduced `chain` and `farcaster` schemes.
+
+```
+<protocol>://<resource>,  <= 2048 characters
+
+protocol      =  "https" | "http" | "ipfs" | "chain" | "farcaster"
+resource      = (defined by each protocol specification)
+```
+
+The Farcaster URL structure adds the following additional protocols to the URL prefix:
+
+- `chain://`
+- `farcaster://`
+
+### Chain Resources
```

Comment:
Per discussion, we should also call out the explicit resources that clients have "soft agreed" to support. Today, that would be eth addresses, eth transactions (maybe down to the index level), erc-20, erc-721 and erc-1155's.